testing/test_capture.py

```diff
@@ -642,6 +642,13 @@ def test_unicode_and_str_mixture(self):
             f.close()
             assert isinstance(s, unicode)
 
+    def test_write_bytes_to_buffer(self):
+        f = capture.CaptureIO()
+        # In python3, stdout / stderr are text io wrappers (exposing a buffer
+        # property of the underlying bytestream).
+        getattr(f, 'buffer', f).write(b'foo\n')
```

Comment:
This seems like it'd pass even if the object doesn't have a `buffer` attribute (i.e. without any code changes).